src/main/java/se/liu/ida/hefquin/engine/queryplan/utils/TextBasedLogicalPlanPrinterImpl2.java

```diff
@@ -0,0 +1,314 @@
+package se.liu.ida.hefquin.engine.queryplan.utils;
+
+import java.io.PrintStream;
+
+import se.liu.ida.hefquin.engine.queryplan.logical.LogicalPlan;
+import se.liu.ida.hefquin.engine.queryplan.logical.LogicalPlanVisitor;
+import se.liu.ida.hefquin.engine.queryplan.logical.LogicalPlanWalker;
+import se.liu.ida.hefquin.engine.queryplan.logical.impl.*;
+import se.liu.ida.hefquin.engine.utils.IndentingPrintStream;
+
+public class TextBasedLogicalPlanPrinterImpl2 implements LogicalPlanPrinter
+{
+	// The string represents '|  '.
+	protected final String levelIndentBase = "\u2502  ";
+	// The string represents '├──'.
+	protected final String nonLastChildIndentBase = "\u251C\u2500\u2500";
+	// The string represents '└──'.
+	protected final String lastChildIndentBase = "\u2514\u2500\u2500";
```

Comment:
You can make these `static` as well. No need to allocate extra memory for every instance of this class.